testing/nightly/custom.sh

```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/env bash
+
+set -o pipefail
+set -o errexit
+set -o nounset
+
+# This script is executed from `ci-artifacts` PR, when requesting
+# `\test custom`.
+#
+# It looks at the message of the last commit of the PR (2nd parent of
+# HEAD merge commit), and scans it for a test-path directive:
+# > test-path: testing/.../test.sh [flags]
+# and executes the test command.
+#
+# Calling this script on a commit without a `test-path` fails the test
+# (exit 1).
+
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd ${THIS_DIR}/../..
+
+ANCHOR="test-path: "
+
+commit="HEAD"
+parent=$(git log --pretty=%P -n 1 $commit)
+
+if grep -q " " <<< "$parent"; then
+    commit=$(cut -d" " -f2 <<< "$parent")
+    echo "HEAD is a merge commit. Taking the 2nd parent from $parent"
+else
+    echo "HEAD is a simple commit."
+fi
+
+git show --quiet "$commit"
+
+echo ""
+
+testpaths=$(git log --format=%B -n 1 $commit | { grep -i "$ANCHOR" || true ;} | cut -b$(echo "$ANCHOR" | wc -c)-)
+
+if [[ -z "$testpaths" ]]; then
+    echo "Nothing to test in $commit."
+    exit 1
+fi
+
+BASE_ARTIFACT_DIR=${ARTIFACT_DIR}
+
+while read cmd;
+do
+    if ! grep "^testing/" -q <<< "$cmd"; then
+        echo "ERROR: test command must start with testing/: $cmd"
+        continue
+    fi
+
+    if grep ";" -q <<< "$cmd"; then
```

Comment:
This is definitely insufficient from a security perspective, there are a lot of ways to bypass this. This is a battle not worth fighting - we must make sure only authorized users can have CI run on their PRs

Reply:
this block is not necessary anymore, I removed the `eval` function that used to be called, and use instead:
```
bash $(echo "$cmd") 
```
my goal was that arbitrary code could not be executed from the commit message, eg
```
test-path: rm -rf /
```
fails with
```
/usr/bin/rm: /usr/bin/rm: cannot execute binary file
```
so only files part of the repo can get executed, and hence already in the repo or part of the PR diff

I'll remove this block